src/main/kotlin/ecommerce/service/AuthService.kt

```diff
@@ -0,0 +1,54 @@
+package ecommerce.service
+
+import ecommerce.dto.MemberDto
+import ecommerce.dto.TokenRequest
+import ecommerce.dto.TokenResponse
+import ecommerce.exception.NotFoundException
+import ecommerce.exception.UnauthorizedException
+import ecommerce.infrastructure.JwtTokenProvider
+import ecommerce.model.Member
+import ecommerce.repository.MemberRepository
+import org.springframework.stereotype.Service
+
+@Service
+class AuthService(
+    private val jwtTokenProvider: JwtTokenProvider,
+    private val memberRepository: MemberRepository,
+) {
+    fun findMember(payload: String): MemberDto {
+        val member = memberRepository.findByEmail(payload)
+        if (member == null) {
+            throw NotFoundException("Member not found")
+        }
+        return MemberDto(member.id!!, member.email, member.role)
+    }
+
+    fun findMemberByToken(token: String): MemberDto {
+        if (!jwtTokenProvider.validateToken(token)) {
+            throw UnauthorizedException("Invalid token")
+        }
+        val payload = jwtTokenProvider.getPayload(token)
+        return findMember(payload)
+    }
```

Comment:
how about ask to jwtTokenProvider?
`if (jwtTokenProvider.isInvalid(token)) { ... }`

or throw exception it self in `getPayload` method.